src/ServiceManagerBase.sol

```diff
@@ -0,0 +1,72 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.12;
+
+import {OwnableUpgradeable} from "@openzeppelin-upgrades/contracts/access/OwnableUpgradeable.sol";
+
+import {ISignatureUtils} from "eigenlayer-contracts/src/contracts/interfaces/ISignatureUtils.sol";
+import {IDelegationManager} from "eigenlayer-contracts/src/contracts/interfaces/IDelegationManager.sol";
+
+import {IServiceManager} from "src/interfaces/IServiceManager.sol";
+import {IRegistryCoordinator} from "src/interfaces/IRegistryCoordinator.sol";
+
+/**
+ * @title Minimal implementation of a ServiceManager-type contract.
+ * This contract can inherited from or simply used as a point-of-reference.
+ * @author Layr Labs, Inc.
+ */
+contract ServiceManagerBase is IServiceManager, OwnableUpgradeable {
+    address immutable registryCoordinator;
+    IDelegationManager immutable delegationManager;
+
+    /// @notice when applied to a function, only allows the RegistryCoordinator to call it
+    modifier onlyRegistryCoordinator() {
+        require(
+            msg.sender == address(registryCoordinator),
+            "ServiceManagerBase.onlyRegistryCoordinator: caller is not the registry coordinator"
+        );
+        _;
+    }
+
+    /// @notice Sets the (immutable) `registryCoordinator` address
+    constructor(
+        IDelegationManager _delegationManager,
+        IRegistryCoordinator _registryCoordinator
+    ) {
+        delegationManager = _delegationManager;
+        registryCoordinator = address(_registryCoordinator);
+        _disableInitializers();
+    }
+
+    function initialize(address initialOwner) external initializer {
+        _transferOwnership(initialOwner);
+    }
+
+    /**
+     * @notice Sets the metadata URI for the AVS
+     * @param _metadataURI is the metadata URI for the AVS
+     * @dev only callable by the owner
+     */
+    function setMetadataURI(string memory _metadataURI) public virtual onlyOwner {
```

Comment:
why is this not in the IServiceManager interface?

Reply:
I asked that in the PR description. Lemme just add it real quick; I think if we have the IServiceManager interface at all, this makes sense to be in it.

Reply:
OK, added it in https://github.com/Layr-Labs/eigenlayer-middleware/pull/109/commits/a8008a07656c082ab9e2df0d8c9543b3d0734392